jerry-core/jrt/jrt.h

```diff
@@ -73,114 +80,57 @@
 
-/**
- * Variable that must not be referenced.
- *
- * May be used for static assertion checks.
- */
-extern uint32_t jerry_unreferenced_expression;
-
-extern void __noreturn jerry_assert_fail (const char *, const char *, const char *, const uint32_t);
-extern void __noreturn jerry_unreachable (const char *, const char *, const char *, const uint32_t);
-extern void __noreturn jerry_unimplemented (const char *, const char *, const char *, const uint32_t);
-
 #ifndef JERRY_NDEBUG
-#define JERRY_ASSERT(x) do { if (__builtin_expect (!(x), 0)) { \
-    jerry_assert_fail (#x, __FILE__, __func__, __LINE__); } } while (0)
-#else /* JERRY_NDEBUG */
-#define JERRY_ASSERT(x) do { if (false) { (void)(x); } } while (0)
-#endif /* !JERRY_NDEBUG */
-
-#define JERRY_UNUSED(x) ((void) (x))
+extern void __noreturn jerry_assert_fail (const char *, const char *, const char *, const uint32_t);
+extern void __noreturn jerry_unreachable (const char *, const char *, const uint32_t);
 
-#ifdef JERRY_ENABLE_LOG
-#define JERRY_DLOG(...) jerry_port_log (JERRY_LOG_LEVEL_DEBUG, __VA_ARGS__)
-#define JERRY_DDLOG(...) jerry_port_log (JERRY_LOG_LEVEL_TRACE, __VA_ARGS__)
-#else /* !JERRY_ENABLE_LOG */
-#define JERRY_DLOG(...) \
+#define JERRY_ASSERT(x) \
   do \
   { \
-    if (false) \
+    if (unlikely (!(x))) \
     { \
-      jerry_ref_unused_variables (0, __VA_ARGS__); \
+      jerry_assert_fail (#x, __FILE__, __func__, __LINE__); \
     } \
   } while (0)
-#define JERRY_DDLOG(...) JERRY_DLOG (__VA_ARGS__)
-#endif /* JERRY_ENABLE_LOG */
-
-#define JERRY_ERROR_MSG(...) jerry_port_log (JERRY_LOG_LEVEL_ERROR, __VA_ARGS__)
-#define JERRY_WARNING_MSG(...) jerry_port_log (JERRY_LOG_LEVEL_WARNING, __VA_ARGS__)
 
-/**
- * Mark for unreachable points and unimplemented cases
- */
-extern void jerry_ref_unused_variables (void *, ...);
-
-#ifndef JERRY_NDEBUG
 #define JERRY_UNREACHABLE() \
   do \
   { \
-    jerry_unreachable (NULL, __FILE__, __func__, __LINE__); \
-  } while (0)
-
-#define JERRY_UNIMPLEMENTED(comment) \
-  do \
-  { \
-    jerry_unimplemented (comment, __FILE__, __func__, __LINE__); \
-  } while (0)
-
-#define JERRY_UNIMPLEMENTED_REF_UNUSED_VARS(comment, ...) \
-  do \
-  { \
-    if (false) \
-    { \
-      jerry_ref_unused_variables (0, __VA_ARGS__); \
-    } \
-    jerry_unimplemented (comment, __FILE__, __func__, __LINE__); \
+    jerry_unreachable (__FILE__, __func__, __LINE__); \
   } while (0)
 #else /* JERRY_NDEBUG */
-#define JERRY_UNREACHABLE() \
-  do \
-  { \
-    jerry_unreachable (NULL, NULL, NULL, 0); \
-  } while (0)
-
-#define JERRY_UNIMPLEMENTED(comment) \
-  do \
-  { \
-    jerry_unimplemented (comment, NULL, NULL, 0); \
-  } while (0)
-
-#define JERRY_UNIMPLEMENTED_REF_UNUSED_VARS(comment, ...) \
-  do \
-  { \
-    if (false) \
-    { \
-      jerry_ref_unused_variables (0, __VA_ARGS__); \
-    } \
-    jerry_unimplemented (comment, NULL, NULL, 0); \
-  } while (0)
+#define JERRY_ASSERT(x) do { if (false) { JERRY_UNUSED (x); } } while (0)
```

Comment:
You one-lined a macro here, but expanded a macro at the end. I think we should use the same style everwhere, and I prefer the later.